docs/concepts.md

```diff
@@ -67,6 +68,96 @@ that are specific to a particular Node (like per-process UNIX file descriptors),
 so handle value 3 for Node A identify a different channel than handle value 3
 for Node B (unless there happens to be a coincidence of numbering).
 
+## Labels
+
+Labels are used to enforce information flow control (IFC) between nodes and
+channels in an Oak Application and the outside world.
+
+Each node and each channel is associated with a label at creation time. The
+label does not change any more after that (apart from explicit declassification
+operations, which are not currently supported).
+
+A label has two components: **secrecy** and **integrity**.
+
+Each component is represented as an (unordered) set of tags, and each tag refers
+to individual security principal in the system.
+
+The following types of security principals are supported in Oak:
+
+- **user**: specified by a bearer token, public key, or some other assertion of
+  a user identity
+- **computation**: specified by a measurement (hash) of the WebAssembly module
+  running within an Oak Node
+
+The least privileged label is "public_trusted" (also referred to as `⊥`,
+pronounced "bottom"), which corresponds to a node or channel which has only
+observed public data, and its inputs are not endorsed with any level of
+integrity; in this label, both secrecy and integrity components are empty sets.
+
+Given two labels `L_a` and `L_b`, `L_a ⊑ L_b` (pronounced "flows to") iff `L_a`
+is at least as strict as `L_b`, i.e. `b` has "more secrecy" and "less integrity"
+than `a`.
+
+Intuitively, data can only flow to nodes or channels with:
+
+- **at least** that level of secrecy
+- **at most** that level of integrity
+
+For instance, if integrity of `a` is `{s_0, s_1}`, and integrity of `a` is
+`{i_0, i_1}`, then:
+
+- data can flow from `a` to `b` if `b`'s secrecy is `{s_0, s_1, s_2}`, since it
+  corresponds to "more secret" data
+- data cannot flow from `a` to `b` if `b`'s secrecy is `{s_0}`, since it
+  corresponds to "less secret" data
+- data cannot flow from `a` to `b` if `b`'s integrity is `{i_0, i_1, i_2}`,
+  since it corresponds to "more endorsed" data
+- data can flow from `a` to `b` if `b`'s integrity is `{i_0}`, since it
+  corresponds to "less endorsed" data
```

Comment:
Similar to my comment above on the phrase endorsed. Instead try:

> since 'a' is trusted by all the integrity tags in b

or

> since a is at least as trusted as b.

Reply:
Done.